lib/rex/ui/text/input/readline.rb

```diff
@@ -187,6 +199,75 @@ def readline_with_output(prompt, add_history=false)
       end
     end
 
+    # Try getting multi-line input support provided by Reline, fall back to Readline.
+    def _multiline_with_fallback
+      query = _multiline
+      query = _fallback if query[:status] == :fail
+
+      query
+    end
+
+    def _multiline
+      begin
+        require 'reline' unless defined?(::Reline)
+        self.extend(::Reline) unless self.kind_of?(::Reline)
+      rescue ::LoadError => e
+        elog('Failed to load Reline', e)
+        return { status: :fail, errors: [e] }
+      end
+
+      stop_words = %w[stop s exit e end quit q].freeze
+
+      finished = false
+      begin
+        prompt_proc_before = ::Reline.prompt_proc
+        ::Reline.prompt_proc = proc { |line_buffer| line_buffer.each_with_index.map { |_line, i| i > 0 ? 'SQL *> ' : 'SQL >> ' } }
```

Comment:
`lib/rex/ui/text/input/readline` is meant to be a reusable module shared by multiple modules, and used in multiple contexts. I don't think it makes sense to have any of this SQL-specific logic here - as it stops the method from be used in any other scenario other than SQL prompts 👀